source/AMDSMainCommon.h

```diff
@@ -14,10 +14,14 @@ namespace AMDSMain_Common {
 	/*
 	 * Print the usage of the application
 	 */
-	void printUsage(const QString &appName)
+	void printUsage(const QString &appName, const QString &errMsg="")
 	{
+		if (errMsg != "") {
+			qDebug() << "\nERROR: " << errMsg << "\n";
+		}
+
 		qDebug() << "\nUsages:\n"
-				 << QString("	%1 --intf=[interface name] [--port=[portnumber]]\n").arg(appName)
+				 << QString("	%1 --intf=[interface name] [--port=[portnumber]] [--cfg=[cfgFileName]]\n").arg(appName)
```

Comment:
yes, --cfg is optional. Ln39 gives the cfgFileName a default value, which doesn't hurt anything. But in ln58, we had a general check for all the required parameters.

But for AMDSPV, this parameter is required. I had a specific check in https://github.com/acquaman/AcquamanDataServer/blob/3a12ce60ac5da83f6baf8be774790a8931146cfe/source/AMDSMainSGMPV.cpp#L20.